xmppserver/src/main/java/org/jivesoftware/openfire/muc/spi/MUCPersistenceManager.java

```diff
@@ -1627,5 +1644,32 @@ public static void deleteLocalProperty(String subdomain, String name) {
     public static void refreshProperties(String subdomain) {
         propertyMaps.replace(subdomain, new MUCServiceProperties(subdomain));
     }
-    
+
+    /**
+     * Check if a room name is retired for a given service.
+     *
+     * @param roomName the name of the room to check.
+     * @param multiUserChatService the service to check the room name against.
+     * @return true if the room name is retired for the supplied service, false otherwise.
+     */
+    public static boolean isRoomRetired(String roomName, MultiUserChatService multiUserChatService) {
```

Comment:
I'm perfectly happy for an answer say: "that's overkill", but: is it worth caching the retirement state of rooms? Is there a chance that this method is invoked in a loop, or repeatedly? Instead of doing a query per room, it may make sense to load all retirees in a (possibly short-lived) cache. Changes to the retirement status of a room will be rare, and occur only in this class, I think, which would give a good opportunity to keep that cache up-to-date. We do need to make sure such a strategy would work in a cluster (by either making this a clustered cache, or in another way ensure that a change on one cluster node is visible on the other nodes).

Reply:
I think this will have to wait for now.

Reply:
I've raised this as https://igniterealtime.atlassian.net/browse/OF-2934 for someone to pick up in the future.